edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerEnvironment.cs

```diff
@@ -74,7 +75,19 @@ public async Task<ModuleSet> GetModulesAsync(CancellationToken token)
                     ? this.restartManager.ComputeModuleStatusFromRestartPolicy(moduleRuntimeInfo.ModuleStatus, dockerModule.RestartPolicy, moduleState.RestartCount, lastExitTime)
                     : moduleRuntimeInfo.ModuleStatus;
 
-                string image = !string.IsNullOrWhiteSpace(dockerRuntimeInfo.Config.Image) ? dockerRuntimeInfo.Config.Image : dockerModule.Config.Image;
+                // Check to see if the originalImage label is set by the iotedege daemon. This label is set for content trust feature where the image is pulled by digest.
+                // OrignalImageLabelStr contains the image with tag and it is used to handle the mismatch of image with digest during reconcilation.
+                var labels = dockerRuntimeInfo.Config.CreateOptions?.Labels ?? new Dictionary<string, string>();
+                labels.TryGetValue(Core.Constants.Labels.OriginalImage, out string orignalImageLabelStr);
+                if (!string.IsNullOrWhiteSpace(orignalImageLabelStr))
```

Comment:
Are you planning to add tests DockerEnvironmentTest for cases when DockerRuntimeInfo contains the image label set in content trust case?

Reply:
Added them.